### PR TITLE
Created Modal in ProfileForm.tsx, added profile deletion from firebas…

### PR DESCRIPTION
**Pull request by dvala041**
# H1 Account-Deletion-FullStack

## Summary
1. Created changeUserStatus function in ProfileForm.tsx which makes a call to /users/:userid endpoint to delete user from prisma and firebase databases (added ability to delete from firebase database in views.ts)
2. Created a Modal for deleting account in ProfileForm.tsx; when account is deleted the user's session ends, they are rerouted to login page and their account is deleted

## Testing
1. Created dummy accounts to test the endpoint on postman and then on our local code

## Notes
1. Akin noted the importance of having the prisma and firebase deletion in the same promise

### CODE REVIEW

File: frontend/src/components/organisms/ProfileForm.tsx [hunk 1]
```diff
@@ -40,8 +50,102 @@ interface ProfileFormProps {
...
+  //changeUserStatus handles deleting user from prisma and firebase
+  const { mutateAsync: changeUserStatus, error: changeUserStatusError } =
+    useMutation({
+      mutationFn: async () => {
+        const { data } = await api.delete(`/users/${userDetails.id}`);
+        return data;
+      },
+      retry: false,
+      onSuccess: () => {
+        console.log("DELETED");
```
Comment (akinfelami) [inline frontend/src/components/organisms/ProfileForm.tsx:76]: remove log

File: frontend/src/components/organisms/ProfileForm.tsx [hunk 2]
```diff
@@ -40,8 +50,102 @@ interface ProfileFormProps {
...
+const ModalBody = ({ userDetails, handleClose }: ModalBodyProps) => {
+  const {
+    register,
+    handleSubmit,
+    watch,
+    formState: { errors },
+  } = useForm<DeleteAccountFormValues>();
+
+  //changeUserStatus handles deleting user from prisma and firebase
+  const { mutateAsync: changeUserStatus, error: changeUserStatusError } =
```
Comment (akinfelami) [inline frontend/src/components/organisms/ProfileForm.tsx:68]: You should use appropriate naming patterns. I don't think `changeUserStatus` is the way

File: frontend/src/components/organisms/ProfileForm.tsx [hunk 3]
```diff
@@ -40,8 +50,102 @@ interface ProfileFormProps {
...
+      console.log(error);
+    }
+  };
+
+  /**  When delete form is submitted, we will delete user, end their session,
+   * and redirect them to login page
+   */
+  const handleDeleteAccount = async () => {
+    await changeUserStatus();
+    await handleSignOut();
```
Comment (akinfelami) [inline frontend/src/components/organisms/ProfileForm.tsx:101]: Wrap this in a try catch, it is possible that it fails and the user will need some feedback that the action failed

File: frontend/src/components/organisms/ProfileForm.tsx [hunk 4]
```diff
@@ -153,8 +262,27 @@ const ProfileForm = ({ userDetails }: ProfileFormProps) => {
...
+  //   mutationFn: async () => {
+  //     const { data } = await api.delete(`/users/${userid}`);
+  //     return data;
+  //   },
+  //   retry: false,
+  //   onSuccess: () => {
+  //     console.log("DELETED");
+  //   }
+  // });
+
```
Comment (akinfelami) [inline frontend/src/components/organisms/ProfileForm.tsx:275]: remove comments

File: frontend/src/components/organisms/ProfileForm.tsx [hunk 5]
```diff
@@ -275,6 +403,11 @@ const ProfileForm = ({ userDetails }: ProfileFormProps) => {
               Reset changes
             </Button>
           </div>
+          <div className="order-2 sm:order-1">
+            <Button variety="error" type="button" onClick={handleOpen}>
+              Delete account
+            </Button>
+          </div>
```
Comment (akinfelami) [inline frontend/src/components/organisms/ProfileForm.tsx:410]: I saw some experimentation with making the Delete account button disabled until the user enters the right email. I think it's a good idea. Let's keep that

**Review by akinfelami — CHANGES_REQUESTED** [review]
Hi guys, great work on this!

I have added a few comments that need addressing. Please check them out and let me know if there are any questions

**Review by akinfelami — APPROVED** [review]

**Review by akinfelami — APPROVED** [review]

**Review by jasozh — APPROVED** [review]

**Pull request merged**